src/views/ArticleView.vue

```diff
@@ -0,0 +1,351 @@
+<template>
+	<div class="article">
+		<div
+			ref="contentRef"
+			class="content"
+			v-html="contentHtml" />
+		<div class="footer">
+			<!--
+				the footer element is copied from the wikipedia website
+				keeping the same classname and structure to maintain the same styling
+			-->
+			<div v-if="relatedArticles" id="mw-data-after-content">
+				<div class="read-more-container">
+					<h2>Related articles</h2>
+					<ul class="ext-related-articles-card-list">
+						<li
+							v-for="article in relatedArticles"
+							:key="article.title"
+							:title="article.title"
+							class="ext-related-articles-card">
+							<a :href="'#/article/' + article.title" @click="goTo( article.title )">
+								<div
+									class="ext-related-articles-card-thumb"
+									:style="{
+										backgroundImage: 'url(' +
+											article.highlights[ 0 ].image +
+											')'
+									}"
+								/>
+								<div class="ext-related-articles-card-detail">
+									<h3>{{ article.title }}</h3>
+									<p class="ext-related-articles-card-extract">
+										{{ article.highlights[ 0 ].text }}
+									</p>
+								</div>
+							</a>
+						</li>
+					</ul>
+				</div>
+			</div>
+			<ul id="footer-info" class="footer-info hlist">
+				<li id="footer-info-copyright">
+					Content is available under <a
+						rel="nofollow"
+						href="https://creativecommons.org/licenses/by-sa/4.0/deed.en">CC BY-SA 4.0</a>, unless otherwise noted.
+				</li>
+				<ul id="footer-places" class="footer-places hlist hlist-separated">
+					<li id="footer-places-privacy">
+						<a href="https://foundation.wikimedia.org/wiki/Special:MyLanguage/Policy:Privacy_policy">Privacy policy</a>
+					</li>
+					<li id="footer-places-terms-use">
+						<a href="https://mobile.test/wiki/Terms_of_Use">Terms of use</a>
+					</li>
+				</ul>
+			</ul>
+		</div>
+	</div>
+</template>
+
+<script setup>
+import { ref, onMounted } from 'vue';
+import { useRoute } from 'vue-router';
+import { categories, getArticle } from '../data.js';
+const route = useRoute();
+const contentHtml = ref();
```

Comment:
You really need a single ref to put all data as well as a `loaded` flag that you can use to v-show at the top level. Loading sequence is still incorrect.

const data = ref( { loaded: false } );